dhis-2/dhis-services/dhis-service-program-rule/src/main/java/org/hisp/dhis/programrule/action/validation/BaseProgramRuleActionValidator.java

```diff
@@ -193,4 +189,81 @@ private ProgramRuleActionValidationResult handleTrackedEntityAttribute(
 
     return ProgramRuleActionValidationResult.builder().valid(true).build();
   }
+
+  private List<ProgramStage> getRequiredProgramStages(
+      @Nonnull ProgramRuleActionValidationContext validationContext, @Nonnull Program program) {
+    List<ProgramStage> availableStages = validationContext.getProgramStages();
+
+    if (hasAllProgramStages(availableStages, program)) {
```

Comment:
Can we remove this check?
We are already returning an empty list from `fetchMissingStages` if there is nothing to return